### PR TITLE
Added Chevron and Icon

### PR DESCRIPTION
**Pull request by snbehar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRButton.swift [hunk 1]
```diff
@@ -4,10 +4,14 @@ class ACRButton: FlatButton {
     public var backgroundColor: NSColor = .init(red: 0.35216, green: 0.823529412, blue: 1, alpha: 1)
     public var hoverBackgroundColor: NSColor = .linkColor
     public var activeBackgroundColor: NSColor = .gray
-    public var chevron: Bool = true {
+    public var iconImage: String = ""
+    public var iconPos: NSControl.ImagePosition = .imageLeft
+    public var chev: Bool = false {
         didSet {
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRButton.swift:10]: Variable names should be verbose enough.
`iconPosition` `showsChevron` `iconImageName`

File: source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift [hunk 2]
```diff
@@ -25,8 +25,12 @@ open class FlatButton: NSButton, CALayerDelegate {
     internal var containerLayer = CALayer()
     internal var iconLayer = CAShapeLayer()
     internal var alternateIconLayer = CAShapeLayer()
+    internal var chevronLayer = CAShapeLayer()
     internal var titleLayer = CATextLayer()
     internal var mouseDown = Bool()
+    public var ico: Bool = false
+    internal var c = 1
+    internal var a = 1
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift:33]: Here also. Use good names to make it readable.

File: source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift [hunk 3]
```diff
@@ -122,11 +126,17 @@ open class FlatButton: NSButton, CALayerDelegate {
             setupImage()
         }
     }
+    open var chevron: Bool = false {
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift:129]:
```suggestion
    open var showsChevron: Bool = false {
```

File: source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift [hunk 4]
```diff
@@ -122,11 +126,17 @@ open class FlatButton: NSButton, CALayerDelegate {
...
+        didSet {
+            chevronDraw(par: "arrowdown")
+        }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift:132]:
```suggestion
        didSet {
            drawChevron(par: "arrowdown")
        }
```
What's `par` here??

File: source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift [hunk 5]
```diff
@@ -182,59 +195,113 @@ open class FlatButton: NSButton, CALayerDelegate {
...
+                chevronRect.origin.x = round((hSpacing * 2) + titleSize.width + (imageRect.width / 2))
+            }
         case .imageBelow:
-            titleRect.origin.y = 2
-            titleRect.origin.x = round((bounds.width - titleSize.width) / 2)
-            imageRect.origin.y = bounds.height - vSpacing - imageRect.height
-            imageRect.origin.x = round((bounds.width - imageRect.width) / 2)
+            print("image setup below")
+            print(vSpacing)
+            titleRect.origin.y = round(vSpacing * 2.5)
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift:227]: `print` statements can be removed.

File: source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift [hunk 6]
```diff
@@ -182,59 +195,113 @@ open class FlatButton: NSButton, CALayerDelegate {
...
+            titleRect.origin.y = round((containerLayer.bounds.height - titleSize.height) / 2)
+            titleRect.origin.x = round((hSpacing * 2) + imageRect.width)
+            imageRect.origin.y = round((containerLayer.bounds.height - imageRect.height) / 2)
             imageRect.origin.x = hSpacing
+            if chevron {
+                chevronRect.origin.y = round((containerLayer.bounds.height - chevronRect.height) / 2)
+                chevronRect.origin.x = round((hSpacing * 3) + titleSize.width + imageRect.width)
+                if c == 1 {
+                    c = 0
+                    containerLayer.frame.size.width += chevronRect.width
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/FlatButton.swift:246]: What's `c` here? If it's a flag, consider using `Bool`

**Review by blue-stone99 — COMMENTED** [review]
Most of the unused Assets can be removed and I guess we can avoid the changes in `RootViewController.xib`.

**Comment by snbehar** [comment]
> Most of the unused Assets can be removed and I guess we can avoid the changes in `RootViewController.xib`.

**Pull request merged**